File: src/main/java/aurocosh/divinefavor/common/custom_data/player/data/escape_plan/EscapePlanData.java
package aurocosh.divinefavor.common.custom_data.player.data.escape_plan;

import aurocosh.divinefavor.common.lib.GlobalBlockPos;

// The default implementation of the capability. Holds all the logic.
public class EscapePlanData {
    private GlobalBlockPos pos;

    public EscapePlanData() {
        pos = new GlobalBlockPos(0, 0, 0, 1);
    }

    public GlobalBlockPos getGlobalPosition() {
        return pos;
    }

    public void setGlobalPosition(GlobalBlockPos pos) {
        this.pos = pos;
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/custom_data/player/data/spell/escape_plan/EscapePlanData.kt
package aurocosh.divinefavor.common.custom_data.player.data.spell.escape_plan

import aurocosh.divinefavor.common.lib.GlobalBlockPos

// The default implementation of the capability. Holds all the logic.
class EscapePlanData {
    var globalPosition: GlobalBlockPos? = null

    init {
        globalPosition = GlobalBlockPos(0, 0, 0, 1)
    }
}
